src/module.c

```diff
@@ -6704,6 +6735,130 @@ int RM_SetModuleUserACL(RedisModuleUser *user, const char* acl) {
     return ACLSetUser(user->user, acl, -1);
 }
 
+/* Retrieve the user ID of the client connection behind the current context.
+ * The user ID can be used later, in order to check if command, key or channel
+ * can be executed or accessed according to the ACLs rules associated with that user.
+ * The caller can later free the user ID using the function RM_FreeUserID
+ *
+ * On success a RedisModuleUserID object is returned, otherwise
+ * NULL is returned. */
+ RedisModuleUserID *RM_GetCurrentUserID(RedisModuleCtx *ctx) {
+    if (ctx->client->user == NULL) {
+        return NULL;
+    }
+
+    RedisModuleUserID *userid = zmalloc(sizeof(RedisModuleUserID));
+    userid->id = createStringObject(ctx->client->user->name, sdslen(ctx->client->user->name));
+    return userid;
+}
+
+/* Frees a given user ID .*/
+void RM_FreeUserID(RedisModuleUserID *userid) {
```

Comment:
I'm going to assume there is some long conversation about this somewhere, but why are we introducing a new resource called a UserID. In the other module API we just pass in the real username?

Reply:
There was some discussion here: https://github.com/redis/redis/pull/9309#discussion_r685956688
I don't remember why it was decided to make an opaque type rather than use a string. @YaacovHazan @yossigo ?

Reply:
First version was passing clients around, we wanted to decouple that but did not want to hold references to RedisModuleUser. Passing a string was an option but an opaque type seemed like a cleaner API that is less prone to abuse and more future proof.

Reply:
So, my understanding is that this should apply to both RedisModuleUsers (which are not owned by the ACL system) and users owned by the ACL system. 

Is there also the intention that modules can hold onto the ID and then pass it to a background thread for async processing? I assume that is why it's not actually holding on to the user.

Reply:
RedisModuleUserID dropped